actions/user_actions.jsx

```diff
@@ -422,6 +424,25 @@ export function autocompleteUsers(username) {
     };
 }
 
+export function autocompleteUsersPrioritizeDM(username) {
+    return async (doDispatch, doGetState) => {
+        const state = doGetState();
+        const directChannels = getAllDirectChannels(state);
+        const sortByLastPost = (a, b) => {
+            return a.last_post_at > b.last_post_at ? -1 : 1;
+        };
+        const directMessageChannels = directChannels.filter((channel) => channel.type === Constants.DM_CHANNEL).sort(sortByLastPost);
+
+        const re = new RegExp(username);
+        const directMessageUsers = directMessageChannels.map((dm) => getUser(state, dm.teammate_id)).filter((user) => re.test(user.first_name) || re.test(user.last_name) || re.test(user.nickname) || re.test(user.username));
```

Comment:
Instead of using a regex, could we maybe use `user.username.startsWith(usernamePrefix)` and such instead? That way, we don't need to worry about the extra overhead of using a regex